src/libstore/tests/derivation.cc

```diff
@@ -89,28 +83,20 @@ TEST_F(DynDerivationTest, BadATerm_oldVersionDynDeps) {
                 mockXpSettings);                                          \
             DerivationOutput expected { VAL };                            \
             ASSERT_EQ(got, expected);                                     \
-        }                                                                 \
+        });                                                               \
     }                                                                     \
                                                                           \
     TEST_F(FIXTURE, DerivationOutput_ ## NAME ## _to_json) {              \
-        auto file = goldenMaster("output-" #NAME ".json");                \
-                                                                          \
-        json got = DerivationOutput { VAL }.toJSON(                       \
-            *store,                                                       \
-            DRV_NAME,                                                     \
-            OUTPUT_NAME);                                                 \
-                                                                          \
-        if (testAccept())                                                 \
-        {                                                                 \
-            createDirs(dirOf(file));                                      \
-            writeFile(file, got.dump(2) + "\n");                          \
-            GTEST_SKIP() << updatingGoldenMaster;                         \
-        }                                                                 \
-        else                                                              \
-        {                                                                 \
-            auto expected = json::parse(readFile(file));                  \
-            ASSERT_EQ(got, expected);                                     \
-        }                                                                 \
+        writeTest<json>("output-" #NAME ".json", [&]() -> json {          \
+            return DerivationOutput { VAL }.toJSON(                       \
+                *store,                                                   \
+                DRV_NAME,                                                 \
+                OUTPUT_NAME);                                             \
+        }, [&](auto & file) {                                             \
+            return json::parse(readFile(file));                           \
+        }, [&](auto & file, auto & got) {                                 \
```

Comment:
What exactly do those lambda expressions capture? If they are not needed, then please leave out the `&` from its `[&]` header. Or do they capture the `this` pointer? If so, it would be helpful to make this more explicit.

Reply:
Nah no capturing needed
```suggestion
        }, [](auto & file, auto & got) {                                 \
```